### PR TITLE
run client/server sample in CI

### PR DESCRIPTION
**Pull request by tmakatos**
Signed-off-by: Thanos Makatos <thanos.makatos@nutanix.com>

### CODE REVIEW

**Comment by tmakatos** [comment]
The number of iterations in the stop-and-copy phase is sometimes 2 and sometimes 3, sounds like a race condition, need to figure out what's wrong.

File: test/test-client-server.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+set -e
+
+sock="/tmp/vfio-user.sock"
+rm -f ${sock}*
+../samples/server -v ${sock} &
+while [ ! -S ${sock} ]; do
+	sleep 0.1
+done
+../samples/client ${sock} &
+wait $!
```
Comment (jlevon) [inline test/test-client-server.sh:12]: shouldn't this just be:

wait

Comment (tmakatos) [reply test/test-client-server.sh:12]: The problem with a simple `wait` is that it return the exit code of the last child that exited, so any other children that failed won't cause `wait` to fail, so the CI will succeed. In my tests it was usually the client that was failing. I'll try to find a way to make the whole script fail even if one child fails.

Comment (tmakatos) [reply test/test-client-server.sh:13]: fixed, take another look

Comment (jlevon) [reply test/test-client-server.sh:13]: why do we even need the wait $! ? Can't we run the client synchronously?

Comment (tmakatos) [reply test/test-client-server.sh:13]: That is a very good point!

Comment (tmakatos) [reply test/test-client-server.sh]: Fixed, please take another look.

**Review by jlevon — APPROVED** [review]

**Pull request merged**